app/src/main/java/com/hyperether/getgoing/ui/fragment/TrackingFragment.java

```diff
@@ -109,32 +116,51 @@
     private Context classContext;
 
     private DbRoute updatedRoute;
+    MapFragment mapFragment;
 
-    @Override
-    protected void onCreate(Bundle savedInstanceState) {
+    public TrackingFragment() {
+        // Required empty public constructor
+    }
 
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
-        // Keep screen on all the time
-        setContentView(R.layout.activity_location);
+        getActivity().getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
+        OnBackPressedCallback callback = new OnBackPressedCallback(true /* enabled by default */) {
+            @Override
+            public void handleOnBackPressed() {
+                onAnyBackButtonPressed();
+            }
+        };
+        requireActivity().getOnBackPressedDispatcher().addCallback(this, callback);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+        return inflater.inflate(R.layout.fragment_tracking, container, false);
+    }
 
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        navigationController = Navigation.findNavController(view);
         mRouteAlreadySaved = true;
 
         // Open the shared preferences
-        mPrefs = getSharedPreferences("SharedPreferences", Context.MODE_PRIVATE);
+        mPrefs = getActivity().getSharedPreferences("SharedPreferences", Context.MODE_PRIVATE);
         // Get a SharedPreferences editor
         mEditor = mPrefs.edit();
         // Get user data shared prefs
-        cbPrefs = getSharedPreferences("CBUserDataPref.txt", Context.MODE_PRIVATE);
+        cbPrefs = getActivity().getSharedPreferences("CBUserDataPref.txt", Context.MODE_PRIVATE);
```

Comment:
@zukovicfilip why do we need activity to access shared preferences? Context should be enough.